app/controllers/actions/GroupAction.scala

```diff
@@ -55,7 +56,28 @@ class GroupAction @Inject()
                                        appConfig: AppConfig): Future[Result] = {
     authAction.isAuthorisedAgent { arn =>
       isOptedIn(arn) { _ =>
-        groupService.getGroup(groupId).flatMap(_.fold(groupNotFound)(body(_)))
+        groupService
+          .getGroup(groupId)
+          .flatMap(_.fold(groupNotFound)(body(_)))
+      }
+    }
+  }
+
+  def withGroupAndPageOfClientsForAuthorisedOptedAgent(groupId: String, page: Int = 1, pageSize: Int = 10)
+                                                      (body: (GroupSummary, PaginatedList[DisplayClient], Arn) => Future[Result])
+                                                      (implicit ec: ExecutionContext,
+                                                       hc: HeaderCarrier,
+                                                       request: MessagesRequest[AnyContent],
+                                                       appConfig: AppConfig): Future[Result] = {
+    authAction.isAuthorisedAgent { arn =>
+      isOptedIn(arn) { _ =>
+        groupService
+          .getCustomSummary(groupId)
+          .flatMap(_.fold(groupNotFound)(summary => {
+            groupService.getPaginatedClientsForCustomGroup(groupId)(page, pageSize)
```

Comment:
This is where I think you'll want the paginated clients from the agency rather than just a group, but also need a way of knowing who on the page is already in the group...

Reply:
yeah. a bit to think about here really.  the APBE/AUCD separation is a bit of a drag . would be good to move AUCD behind APBE

Reply:
Agreed, that's what I had in mind - ideally APBE should do the heavy lifting here. Hopefully that'll work, but we'll see